packages/astro/src/core/build/graph.ts

```diff
@@ -29,7 +29,7 @@ export function* getTopLevelPages(
 ): Generator<string, void, unknown> {
 	for (const info of walkParentInfos(id, ctx)) {
 		const importers = (info?.importers || []).concat(info?.dynamicImporters || []);
-		if (importers.length <= 2 && importers[0] === resolvedPagesVirtualModuleId) {
```

Comment:
This assumed a `pages` entry would never be imported, which breaks `glob` imports for MD files in `/pages`. The `?mdImport` flag hid this bug before!

Reply:
I fixed this a different way in https://github.com/withastro/astro/pull/4156

Reply:
Oh, glad to hear it! Should I revert and let your PR go first @matthewp ?

Reply:
Update: Rebased with your fix 🙏

Reply:
Yep, sorry I forgot to reply here.